src/dpp/integration.cpp

```diff
@@ -116,7 +116,7 @@ bool integration::expiry_kicks_user() const {
 	return flags & if_expire_kick;
 }
 
-connection::connection() : id(0), revoked(false), verified(false), friend_sync(false), show_activity(false), visible(false) {
+connection::connection() : id({}), revoked(false), verified(false), friend_sync(false), show_activity(false), visible(false) {
```

Comment:
i'm not sure this is more readable? Wouldnt `0ULL` be nicer than `{}`?

Reply:
Initializing a string with `0` is undefined behaviour since 0 gets implicitly converted to `nullptr` and initializing a std::string is UB.

This has always been an issue but C++23 finally now adds a compile time check for it. `{}` makes the string be default initialized instead. If you want, you can completely remove that initialization as well

Reply:
id isnt a string is it? its a dpp::snowflake which is container for uint64_t?

Reply:
https://github.com/brainboxdotcc/DPP/blob/master/include/dpp/integration.h#L140

In the connection class it is a string